### ISSUE TITLE
Rules with suggestions must set the `meta.hasSuggestions` property to `true`

### ISSUE THREAD

**Issue by techandmedia** [nrwl/nx#8660]
When running 
```bash
nx run ui-components-v1:lint
// error
Linting "ui-components-v1"...
Rules with suggestions must set the `meta.hasSuggestions` property to `true`.
Occurred while linting D:\Project\React UI Component\ui-react-component\libs\ui-components-v1\src\lib\phone-number\phonhone-number.tsx:40
Rule: "react-hooks/exhaustive-deps"
```

report
```bash
Node : 16.13.1
  OS   : win32 x64
  npm  : 8.1.2

  nx : 13.4.6
  @nrwl/angular : undefined
  @nrwl/cli : 13.4.6
  @nrwl/cypress : 13.4.6
  @nrwl/devkit : 13.4.6
  @nrwl/eslint-plugin-nx : 13.4.6
  @nrwl/express : undefined
  @nrwl/jest : 13.4.6
  @nrwl/linter : 13.4.6
  @nrwl/nest : undefined
  @nrwl/next : undefined
  @nrwl/node : undefined
  @nrwl/nx-cloud : undefined
  @nrwl/react : 13.4.6
  @nrwl/react-native : undefined
  @nrwl/schematics : undefined
  @nrwl/tao : 13.4.6
  @nrwl/web : 13.4.6
  @nrwl/workspace : 13.4.6
  @nrwl/storybook : 13.4.6
  @nrwl/gatsby : undefined
  typescript : 4.4.4
  rxjs : 6.6.7
  ---------------------------------------
  Community plugins:

(node:11868) [DEP0148] DeprecationWarning: Use of deprecated folder mapping "./" in the "exports" field module resolution of the package at D:\Project\React UI Component\ui-react-component\node_modules\tslib\package.json.
Update this package.json to use a subpath pattern like "./*".
(Use `node --trace-deprecation ...` to show where the warning was created)
```

**Comment by FrozenPandaz** [comment]
Could you please provide a repo that we can pull down to reproduce and debug the issue?

**Comment by techandmedia** [comment]
Hi, so I created a new project from scratch and the error gone.
https://github.com/techandmedia/nx-reporting/tree/error/rules-with-suggestions-must-set-the-meta-hasSuggestions-property-to-true

I checkhed that "eslint-plugin-react-hooks": "4.3.0" is installed which might fix the issue, in my project I used v 4.2 which I then upgraded to 4.2.1-rc and the issue gone

**Comment by meeroslav** [comment]
@techandmedia thank you for reporting this issue. Unfortunately, in some scenarios (partial updates) dependencies go out of sync causing these unexpected issues.

I will close this issue, as it seems to be fixed by just reinstalling the dependencies but it will still be in the history if someone else stumbles upon it.
